gammapy/spectrum/cosmic_ray.py

```diff
@@ -55,17 +33,53 @@ def cosmic_ray_flux(energy, particle="proton"):
     flux : `~astropy.units.Quantity`
         Cosmic ray flux in unit ``m^-2 s^-1 TeV^-1 sr^-1``
     """
-    pars = {
-        "electron": {"N": 6.85e-5, "k": 3.21, "L": 3.19e-3, "E_p": 0.107, "w": 0.776},
-        "proton": {"N": 0.096, "k": 2.70},
-        "N": {"N": 0.0719, "k": 2.64},
-        "Si": {"N": 0.0284, "k": 2.66},
-        "Fe": {"N": 0.0134, "k": 2.63},
+
+    proton = {
+        "amplitude": 0.096 * u.Unit("1 / (m2 s TeV sr)"),
+        "index": 2.70,
+        "reference": 1 * u.TeV,
+    }
+
+    N = {
+        "amplitude": 0.0719 * u.Unit("1 / (m2 s TeV sr)"),
+        "index": 2.64,
+        "reference": 1 * u.TeV,
     }
 
-    if particle == "electron":
-        return _electron_spectrum(energy, **pars["electron"])
-    elif particle in ["proton", "He", "N", "Si", "Fe"]:
-        return _power_law(energy, **pars[particle])
+    Si = {
+        "amplitude": 0.0284 * u.Unit("1 / (m2 s TeV sr)"),
+        "index": 2.66,
+        "reference": 1 * u.TeV,
+    }
+
+    Fe = {
+        "amplitude": 0.0134 * u.Unit("1 / (m2 s TeV sr)"),
+        "index": 2.63,
+        "reference": 1 * u.TeV,
+    }
+
+    electron_power_law = {
+        "amplitude": 6.85e-5 * u.Unit("1 / (m2 s TeV sr)"),
+        "index": 3.21,
+        "reference": 1 * u.TeV,
+    }
+    electron_lognormal = {
+        "norm": 3.19e-3 * u.Unit("1 / (m2 s sr)"),
+        "mean": 0.107 * u.TeV,
+        "sigma": 0.776,
+    }
+
+    if particle == "proton":
+        model = PowerLaw(**proton)
```

Comment:
I would just `return` the model from each case, instead of storing it in a local variable and returning that `model` at the end. Saves one line at the end. But that's a matter of taste. If you prefer like this, OK.